4/README.md

```diff
@@ -0,0 +1,207 @@
+- docker-compose.network.yml
+  - Used to deploy 4 node development environment for BigchainDB(integrated with Tendermint)
+- docker-compose.docs.yml
+  - Used to build and host BigchainDB docs.
+- docker-compose.ssl.yml
+  - DEPRECATED: Used for deployment of single node BigchainDB v1.3, SSL enabled for MongoDB authentication and authorization.
+- docker-compose.rdb.yml
+  - DEPRECATED: Used to deploy single node BigchainDB v1.3 with RethinkDB as the backend database.
+- docker-compose.benchmark.yml
+  - DEPRECATED: Used to deploy standalone BigchainDB v1.3 development environment.
+
+
+Majority of the above mentioned files are either not in use OR can be integrated in a way that all of them are being tested effectively with CI pipeline(s).
+
+We also have multiple Dockerfiles in the repository:
+
+- Dockerfile
+  - Dockerfile used to build and publish BigchainDB production images.
+- Dockerfile-dev
+  - Dockerfile used for development purposes.
+- compose/bigchaindb-driver/Dockerfile
+  - Dockerfile used to spawn a container in development environment with bigchaindb-driver installed.
+    - This seems to be an invalid usecase for a separate container. In my opinion, `bigchaindb-driver` does not qualify as a service. It is just a python-package that can reside on a any of the containers, preferably BigchainDB.
+- compose/bigchaindb-server/Dockerfile
+  - This is a duplicate of `Dockerfile-dev`
+- compose/travis/Dockerfile
+  - Dockerfile used to build/run BigchainDB service in CI.
+
+## Motivation
+There are multiple problems with our current workflow and directory structure which also impacts testing:
+
+- We need to deprecate or remove the unused or redundant docker-compose and Docker file(s).
+- Since our development and CI files are different, there is no way to test the development `docker-compose` file(s). There are two solutions to this:
+  - Write tests to test the docker-compose for development.
+  - Integrate the development and CI workflow.
+
+
+## Specification
+
+### Proposed Solution
+
+- Clean up the deprecated workflow(s).
+  - Remove `docker-compose.rdb.yml`.
+  - Remove `docker-compose.benchmark.yml`.
+  - Remove `docker-compose.docs.yml`, since building documentation should be part of a development environment.
+  - Remove `docker-compose.ssl.yml`, because we are planning to remove SSL support for MongoDB.
+  - Remove `docker-compose.network.yml` file, this workflow will be supported by another deployment method. Currently, we are not using this method for development. We will have Vagrant, Ansible and K8s network driver to offer this functionality.
+
+- Update the workflows with the new deployment workflow(s).
+  - Update `docker-compose.yml` to support the current deployment strategy i.e. BigchainDB + Tendermint + Stand alone MongoDB.
+- Integrate the development and CI compose file(s) in such a way that there are minimal differences and they can be tested on a CI, also rename the files:
```

Comment:
Why can't we use the same file for CI and dev?

Reply:
I mentioned this later, because the dev will have `bigchaindb-driver`, `nginx`, `curl-client` which is used for dev only and will be an overkill for CI.

Reply:
In that case we can still keep them in the same file without CI needing to use them. e.g. `bdb` can have `mdb` and `tendermint` as dependendant services and `bigchaindb-driver` `nginx` `curl-client` can be in the same file but need separate `docker-compose up <service_name>` commands and CI can just not use them. I think it will be simpler without worrying about keeping both `docker-compose` files in sync.